scan/lib/scan/runner.rb

```diff
@@ -80,8 +84,16 @@ def handle_results(tests_exit_status)
     end
 
     def test_results
-      return "" unless Scan.cache[:temp_junit_report]
-      File.read(Scan.cache[:temp_junit_report])
+      temp_junit_report = Scan.cache[:temp_junit_report]
+      return File.read(temp_junit_report) if temp_junit_report && File.file?(temp_junit_report)
```

Comment:
Here we would generate the file for each call to `test_results`.

What are the code paths that lead to the file not being there?

Why can't we reuse the `XCPrettyReporterOptionsGenerator`?